packages/aws/src/RPA/Cloud/AWS/__init__.py

```diff
@@ -990,7 +1024,672 @@ def delete_queue(self, queue_name: str = None):
+                f"an error was encountered: \n{error_message}"
+                f'\n\nFor statement: \n"{query_string}"'
+            ) from e
+
+        finished_statement = client.describe_statement(Id=statement_id)
+        self.logger.info(
+            "Statement finished, total rows affected: "
+            + str(finished_statement.get("ResultRows", "NONE"))
+        )
+        if finished_statement["HasResultSet"]:
+            paginator = client.get_paginator("get_statement_result")
+            full_result = paginator.paginate(Id=statement_id).build_full_result()
+
+            tables = import_tables()
+            if not tables:
+                self.logger.info(
+                    "Tables in the AWS response will be in a `dictionary` type, "
+                    "because `RPA.Tables` library is not available in the scope."
+                )
+            column_names = [
+                m.get("name") for m in full_result.get("ColumnMetadata", {})
+            ]
+            data = [
+                {c: self._parse_tagged_union(f) for c, f in zip(column_names, row)}
+                for row in full_result.get("Records", [])
+            ]
+            return tables().create_table(data) if tables else data
+        else:
+            return finished_statement.get("ResultRows", 0)
+
+    def create_redshift_statement_parameters(self, **params) -> List[Dict[str, str]]:
+        r"""Returns a formatted dictionary to be used in
+        Redshift Data Api SQL statements.
+
+        **Example:**
+
+        Assume the ``${SQL}`` statement has the parameters ``:id`` and
+        ``:name``:
+
+        .. code-block:: robotframework
+
+            *** Tasks ***
+
+            ${params}=    Create sql parameters    id=123    name=Nokia
+            # params produces a data structure like so:
+            #   [
+            #        {"name":"id", "value":"123"},
+            #        {"name":"name", "value":"Nokia"}
+            #    ]
+
+            # Which can be used for the 'parameters' argument.
+            ${response}=    Execute redshift statement    ${SQL}    ${params}
+        """
+        return [{"name": k, "value": v} for (k, v) in params.items()]
+
+    def _parse_tagged_union(self, tagged_union: dict):
+        if tagged_union.get("blobValue"):
+            output = bytes(tagged_union["blobValue"])
+        elif tagged_union.get("booleanValue"):
+            output = bool(tagged_union["booleanValue"])
+        elif tagged_union.get("doubleValue"):
+            output = float(tagged_union["doubleValue"])
+        elif tagged_union.get("isNull"):
+            output = None if tagged_union["isNull"] else "UNKNOWN_DATA_MEMBER"
+        elif tagged_union.get("longValue"):
+            output = int(tagged_union["longValue"])
+        elif tagged_union.get("stringValue"):
+            output = str(tagged_union["stringValue"])
+        else:
+            output = "UNKNOWN_DATA_MEMBER"
+        return output
```

Comment:
Might be more maintainable to have a constant dict here with `{"blobValue": bytes, ...}` and iterate over it to get the values from `tagged_union` associated with their casting types.

Reply:
Great idea, thanks, this is much more clean